packages/rspack/src/generators/application/application.ts

```diff
@@ -1,111 +1,101 @@
+import rspackInitGenerator from '../init/init';
 import { normalizeOptions } from './lib/normalize-options';
-import { ApplicationGeneratorSchema, NormalizedSchema } from './schema';
+import { ApplicationGeneratorSchema } from './schema';
 
 export default async function (
   tree: Tree,
   _options: ApplicationGeneratorSchema
 ) {
-  await ensurePackage(tree, '@nrwl/react', nxVersion);
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const { reactInitGenerator } = require('@nrwl/react');
-
   const tasks = [];
+  const initTask = await rspackInitGenerator(tree, _options);
+  tasks.push(initTask);
+
   const options = normalizeOptions(tree, _options);
 
   options.style ??= 'css';
 
-  addProjectConfiguration(tree, options.name, {
-    root: options.appProjectRoot,
-    projectType: 'application',
-    sourceRoot: `${options.appProjectRoot}/src`,
-    targets: {},
-  });
-
-  const offsetFromRoot = _offsetFromRoot(options.appProjectRoot);
-  generateFiles(tree, path.join(__dirname, 'files'), options.appProjectRoot, {
-    ...options,
-    ...names(options.name),
-    offsetFromRoot,
-    template: '',
-  });
-
-  createTsConfig(
-    tree,
-    options,
-    joinPathFragments(offsetFromRoot, 'tsconfig.base.json')
-  );
-
-  const projectTask = await configurationGenerator(tree, {
-    project: options.name,
-    devServer: true,
-    tsConfig: joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
-    framework: 'react',
-    target: 'web',
-    main: joinPathFragments(options.appProjectRoot, 'src/main.tsx'),
-    newProject: true,
-  });
-  tasks.push(projectTask);
+  if (options.framework === 'nest') {
+    const { applicationGenerator: nestAppGenerator } = ensurePackage(
+      '@nrwl/nest',
+      nxVersion
+    );
+    const createAppTask = await nestAppGenerator(tree, {
+      ...options,
+      skipFormat: true,
+      tags: options.tags ?? '',
+    });
 
-  const jestTask = await addJest(tree, options);
-  tasks.push(jestTask);
+    const convertAppTask = await configurationGenerator(tree, {
+      project: options.name,
+      target: 'node',
+      newProject: false,
+      buildTarget: 'build',
+      framework: 'nest',
+    });
 
-  const cypressTask = await addCypress(tree, options);
-  tasks.push(cypressTask);
-
-  const lintTask = await addLinting(tree, options);
-  tasks.push(lintTask);
-
-  const installTask = addDependenciesToPackageJson(
-    tree,
-    { react: reactVersion, 'react-dom': reactDomVersion },
-    {
-      ...getStyleDependency(options),
-      '@nrwl/react': nxVersion,
-      '@types/react': typesReactVersion,
-      '@types/react-dom': typesReactDomVersion,
-    }
-  );
-  tasks.push(installTask);
-
-  const reactInitTask = await reactInitGenerator(tree, options);
-  tasks.push(reactInitTask);
+    tasks.push(createAppTask, convertAppTask);
+  } else if (options.framework === 'web') {
+    const { applicationGenerator: webAppGenerator } = ensurePackage(
```

Comment:
I think we'll add rspack to `@nrwl/web`, `@nrwl/react` and `@nrwl/nest` for Nx 16 so eventually this app generator won't be used anymore. It's fine for now.

Reply:
Ok!